### ISSUE TITLE
Building LibRI failed on OpenMPI

### ISSUE THREAD

**Issue by caic99** [deepmodeling/abacus-develop#5194]
### Describe the bug

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1040938&view=logs&j=2ed0550f-e734-5ebf-52c6-1bfc19f5811c&t=9c969b19-a14a-5482-8fb9-78bdf745c3c6&l=1332

```

/home/conda/feedstock_root/build_artifacts/abacus_1727661999644/work/deps/LibComm/include/Comm/Comm_Assemble/../Comm_Keys/../global/Cereal_Func.hpp:53:109: error: 'Comm::MPI_Wrapper::char_contiguous' cannot be used as a function
   53 |                 MPI_CHECK( MPI_Send  ( str.c_str(), str.size()>>exponent_align, MPI_Wrapper::char_contiguous(exponent_align), rank_recv, tag, mpi_comm ) );
      |                                                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
/home/conda/feedstock_root/build_artifacts/abacus_1727661999644/work/deps/LibComm/include/Comm/Comm_Assemble/../Comm_Keys/../global/Cereal_Func.hpp:17:26: note: in definition of macro 'MPI_CHECK'
   17 | #define MPI_CHECK(x) if((x)!=MPI_SUCCESS)       throw std::runtime_error(std::string(__FILE__)+" line "+std::to_string(__LINE__));
      |                          ^
/home/conda/feedstock_root/build_artifacts/abacus_1727661999644/work/deps/LibComm/include/Comm/Comm_Assemble/../Comm_Keys/../global/Cereal_Func.hpp: In function 'void Comm::Cereal_Func::mpi_isend(const std::string&, std::size_t, int, int, ompi_communicator_t* const&, ompi_request_t*&)':
/home/conda/feedstock_root/build_artifacts/abacus_1727661999644/work/deps/LibComm/include/Comm/Comm_Assemble/../Comm_Keys/../global/Cereal_Func.hpp:79:110: error: 'Comm::MPI_Wrapper::char_contiguous' cannot be used as a function
   79 |                 MPI_CHECK( MPI_Isend  ( str.c_str(), str.size()>>exponent_align, MPI_Wrapper::char_contiguous(exponent_align), rank_recv, tag, mpi_comm, &request ) );
      |                                                                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
/home/conda/feedstock_root/build_artifacts/abacus_1727661999644/work/deps/LibComm/include/Comm/Comm_Assemble/../Comm_Keys/../global/Cereal_Func.hpp:17:26: note: in definition of macro 'MPI_CHECK'
   17 | #define MPI_CHECK(x) if((x)!=MPI_SUCCESS)       throw std::runtime_error(std::string(__FILE__)+" line "+std::to_string(__LINE__));
      |                          ^
/home/conda/feedstock_root/build_artifacts/abacus_1727661999644/work/deps/LibComm/include/Comm/Comm_Assemble/../Comm_Keys/../global/Cereal_Func.hpp: In function 'std::vector<char> Comm::Cereal_Func::mpi_recv(ompi_communicator_t* const&, MPI_Status&)':
/home/conda/feedstock_root/build_artifacts/abacus_1727661999644/work/deps/LibComm/include/Comm/Comm_Assemble/../Comm_Keys/../global/Cereal_Func.hpp:105:83: error: 'Comm::MPI_Wrapper::char_contiguous' cannot be used as a function
  105 |                         const MPI_Datatype mpi_type = MPI_Wrapper::char_contiguous(exponent_align);
      |                                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
/home/conda/feedstock_root/build_artifacts/abacus_1727661999644/work/deps/LibComm/include/Comm/Comm_Assemble/../Comm_Keys/../global/Cereal_Func.hpp: In function 'std::vector<char> Comm::Cereal_Func::mpi_mrecv(ompi_message_t*&, const MPI_Status&)':
/home/conda/feedstock_root/build_artifacts/abacus_1727661999644/work/deps/LibComm/include/Comm/Comm_Assemble/../Comm_Keys/../global/Cereal_Func.hpp:149:83: error: 'Comm::MPI_Wrapper::char_contiguous' cannot be used as a function
  149 |                         const MPI_Datatype mpi_type = MPI_Wrapper::char_contiguous(exponent_align);
      |                                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~

```

### Environment

OpenMPI 4.1.6
The CXX compiler identification is GNU 12.4.0

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

**Comment by QuantumMisaka** [comment]
@caic99 
#5033

**Comment by PeizeLin** [comment]
Compiling LibComm v0.1.1 using OpenMPI will cause such error.
As a temporary solution, you can compile v0.1.1 using MPICH or IntelMPI, or you can compile v0.1.0 using OpenMPI.

**Comment by PeizeLin** [comment]
fixed in the latest version of [LibComm 2024.10.03-50b98fe](https://github.com/abacusmodeling/LibComm/commit/50b98fe14e4d1cbe929d0306ce9bddb23b3bf975)
